sqlite.go

```diff
@@ -0,0 +1,542 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sqlite implements a database/sql driver for SQLite3.
+//
+// This driver requires a file: URI always be used to open a database.
+// For details see https://sqlite.org/c3ref/open.html#urifilenames.
+//
+//
+// Memory Mode
+//
+// In-memory databases are popular for tests.
+// If opening a database in memory mode with the "?mode=memory" query
```

Comment:
Above it says to always use file://. How does this square with in-memory mode?

Reply:
`file:somefile?mode=memory` opens in-memory mode in SQLite.

Reply:
would it be good to clone how upstream sqlite3 does it? that uses `:memory:`

Reply:
This is how sqlite3 does it. The ":memory:" mode is their pre-URI naming scheme that doesn't let you distinguish different names. In this form, file:whatever?mode=memory is an in-memory table named whatever.

Examples and details are in the SQLite docs linked a few lines earlier. I'm reluctant to repeat their docs here much. I put one example in.